common/dependabot-common.gemspec

```diff
@@ -18,8 +18,8 @@ Gem::Specification.new do |spec|
   spec.require_path = "lib"
   spec.files        = []
 
-  spec.required_ruby_version = ">= 2.7.0"
-  spec.required_rubygems_version = ">= 2.7.3"
+  spec.required_ruby_version = ">= 3.1.0"
+  spec.required_rubygems_version = ">= 3.3.3"
```

Comment:
This is the minimum rubygems version required by ruby `3.1`.

Reply:
I think we'd want to bump it in the Dockerfile in that case? https://github.com/dependabot/dependabot-core/blob/8cdbf1e37096c0ac06ab192fbc07b3f5fc5c85e7/Dockerfile#L74

Reply:
Yep, already bumped it there: https://github.com/dependabot/dependabot-core/pull/5447/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L74

The reason it's bumped here is because otherwise `Rubocop` complains that `spec.required_ruby_version` is not >= to [`TargetRubyVersion`](https://github.com/dependabot/dependabot-core/pull/5447/files#diff-4f894049af3375c2bd4e608f546f8d4a0eed95464efcdea850993200db9fef5cR9)

Reply:
Just  noticed your note about the past update... thx! We'll need to figure out what the root cause was as we _have_ to bump rubygems in order to bump ruby... looks like https://github.com/dependabot/dependabot-core/pull/5044#issuecomment-1109348793 and following discussion is a good place to start.

Reply:
My guess is that with #4973 these issues might no longer happen after upgrading RubyGems.

The rationale is that, if I understood correctly, those issues (and the RubyGems changes that caused them) were related to the implicit Bundler activation mechanism in RubyGems. However, with #4973 Dependabot will no longer rely on that, but will activate Bundler explicitly instead (through `gem "bundler", "~> 1.17"`, for example).

I guess we need to try and see.

Reply:
I think the root cause of the updater specs failing is this bumping of Ruby gems... more specifically as outlined in https://github.com/dependabot/dependabot-core/pull/5513:
> When upgrading RubyGems to 3.3+, RubyGems gets confused about which Bundler version to use when shelling out to native helpers.

Which is exactly the problem I observe in https://github.com/dependabot/dependabot-core/pull/5447#issuecomment-1263989537.

Reply:
I split the ruby gems upgrade out into a separate PR that needs to land before this one: https://github.com/dependabot/dependabot-core/pull/5823

That way it's easier for us to debug any possible issues as Rubygems vs Ruby issues.